src/main/java/uk/gov/justice/probation/courtcaseservice/controller/OffenderMatchesController.java

```diff
@@ -48,28 +53,38 @@ Mono<ResponseEntity<Object>> createGroupedOffenderMatchesByDefendant(@PathVariab
+                                                                  @PathVariable(value = DEFENDANT_ID_PARAM) String defendantId,
+                                                                  @PathVariable(value = GROUP_ID_PARAM) Long groupId) {
         return offenderMatchService.getGroupedMatchesByCaseId(caseId, defendantId, groupId);
     }
 
     @Operation(description = "Returns all possible matches found for a given case and defendant ID")
     @GetMapping(value = "/case/{caseId}/defendant/{defendantId}/matchesDetail", produces = APPLICATION_JSON_VALUE)
     @ResponseStatus(HttpStatus.OK)
     public @ResponseBody
-    OffenderMatchDetailResponse getOffenderMatchesDetailByCaseAndDefendantId(@PathVariable(value = "caseId") String caseId,
-                                                                             @PathVariable(value = "defendantId") String defendantId) {
+    OffenderMatchDetailResponse getOffenderMatchesDetailByCaseAndDefendantId(@PathVariable(value = CASE_ID_PARAM) String caseId,
+                                                                             @PathVariable(value = DEFENDANT_ID_PARAM) String defendantId) {
         return offenderMatchService.getOffenderMatchDetailsByCaseIdAndDefendantId(caseId, defendantId);
     }
 
-
-    @Operation(description = "Gets an existing grouped-offender-matches-entity associated with a defendant and a group")
-    @GetMapping(value = GET_GROUPED_OFFENDER_MATCHES_BY_DEFENDANT_ID_AND_GROUP_ID_PATH, consumes = APPLICATION_JSON_VALUE, produces = APPLICATION_JSON_VALUE)
+    @Operation(description = "Gets an existing grouped-offender-match entity associated with a defendant and a group")
+    @GetMapping(value = GET_GROUPED_OFFENDER_MATCHES_ENTITY_BY_DEFENDANT_ID_AND_GROUP_ID_PATH, consumes = APPLICATION_JSON_VALUE, produces = APPLICATION_JSON_VALUE)
     @ResponseStatus(HttpStatus.OK)
     public @ResponseBody
-    Mono<GroupedOffenderMatchesEntity> getGroupedOffenderMatchesEntityByDefendantIdAndGroupId(@PathVariable(value = "defendantId") String defendantId,
-                                                                                              @PathVariable(value = "groupId") Long groupId) {
-        return offenderMatchService.getGroupedOffenderMatchesEntityByDefendantIdAndGroupId(defendantId, groupId);
+    Mono<GroupedOffenderMatchesEntity> getGroupedOffenderMatchesByDefendantIdAndGroupId(@PathVariable(value = DEFENDANT_ID_PARAM) String defendantId,
+                                                                                        @PathVariable(value = GROUP_ID_PARAM) Long groupId) {
+        return offenderMatchService.getGroupedOffenderMatchesByDefendantIdAndGroupId(defendantId, groupId);
+    }
+
+    @Operation(description = "Creates a new grouped-offender-match entity associated with a defendant")
+    @PostMapping(value = CREATE_GROUPED_OFFENDER_MATCHES_ENTITY_FOR_DEFENDANT_ID_PATH, consumes = APPLICATION_JSON_VALUE, produces = APPLICATION_JSON_VALUE)
```

Comment:
Is there a reason you've extracted the url template as a constant here?

Reply:
Just for better readability.